assets/js/atomic/blocks/product/title/block.js

```diff
@@ -3,30 +3,57 @@
  */
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
-import { useInnerBlockConfigurationContext } from '@woocommerce/shared-context';
 import { decodeEntities } from '@wordpress/html-entities';
+import {
+	useInnerBlockLayoutContext,
+	useProductDataContext,
+} from '@woocommerce/shared-context';
 
+/**
+ * Product Title Block Component.
+ *
+ * @param {Object}  props                Incoming props.
+ * @param {string}  [props.className]    CSS Class name for the component.
+ * @param {number}  [props.headingLevel] Heading level (h1, h2 etc)
+ * @param {boolean} [props.productLink]  Whether or not to display a link to the product page.
+ * @param {Object}  [props.product ]     Optional product object. Product from context will be used if
+ *                                       this is not provided.
+ * @return {*} The component.
+ */
 const ProductTitle = ( {
 	className,
-	product,
 	headingLevel = 2,
 	productLink = true,
+	...props
 } ) => {
-	const { layoutStyleClassPrefix } = useInnerBlockConfigurationContext();
-	if ( ! product.name ) {
-		return null;
+	const productDataContext = { ...useProductDataContext(), ...props };
+	const { layoutStyleClassPrefix } = useInnerBlockLayoutContext();
+	const { product } = productDataContext;
```

Comment:
Personal opinion here, but these two lines were confusing me a bit:
```JS
const productDataContext = { ...useProductDataContext(), ...props };
const { product } = productDataContext;
```
I think this can be rewritten so the spread syntax is not necessary:
```JS
const productDataContext = useProductDataContext();
const { product } = productDataContext || props;
```
I know it's just personal taste, but proposing it in case you like it (it's affecting all other inner blocks). :slightly_smiling_face:

Reply:
@Aljullu This breaks :) Context has value - we just want to take the product from it, or props, with props taking priority.

Reply:
This works

```
const productDataContext = useProductDataContext();
const product = props.product || productDataContext.product;
```